src/java/org/apache/cassandra/repair/RepairSession.java

```diff
@@ -351,12 +368,21 @@ public void onFailure(Throwable t)
                 Tracing.traceRepair("Session completed with the following error: {}", t);
                 forceShutdown(t);
             }
-        }, executor);
+        }, taskExecutor);
     }
 
-    public void terminate()
+    private volatile List<RepairJob> jobs = Collections.emptyList();
```

Comment:
nit: I don't know if we have any hard rules about instance fields being at the start of the class...